pkg/controllers/proxy/network_services_controller.go

```diff
@@ -793,6 +793,48 @@ func (nsc *NetworkServicesController) publishMetrics(serviceInfoMap serviceInfoM
 	return nil
 }
 
+func endpointsListsEquivalent(a, b []endpointsInfo) bool {
+	if len(a) != len(b) {
+		return false
+	}
+
+	for _, epA := range a {
+	        valid := false
+		for _, epB := range b {
+			if epA == epB {
+				valid = true
+				break
+			}
+		}
+
+		if !valid {
+		    return false
+		}
+	}
+	return true
```

Comment:
@ananace thanks so much for your work!

Sorry, that it has taken me so long to come back around to this. However, I think we still need one more change here. Unlike map (hash) keys, lists (slices) aren't guaranteed to be unique, so it's not enough here to only evaluate the size and make sure that all of the endpoints in A exist in B, as A may have some duplicate entries and still be the same size. We should be complete here and also iterate over B and make sure that all of the endpoints in B also exist in A.

While I think that the logic you have here should be sufficient for 99.9% of the use-cases I can think up, and I truly don't want to double-iterate here, working on this project has taught me that there are always edge cases and bugs like this can be hard to track down.

So two options here:
1) Add another function to make the slices have unique values (usually in Go this is done by transferring them to hash keys) before comparing them (if you make them a map you could re-use your existing function below)
2) Duplicate the range function you have here, this time making b the outside loop and a the inside loop

I think option 1 would likely be more performant and would allow you to re-use logic.